src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java

```diff
@@ -0,0 +1,244 @@
+package eu.bbmri_eric.negotiator.service;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import eu.bbmri_eric.negotiator.database.model.AccessForm;
+import eu.bbmri_eric.negotiator.database.model.DiscoveryService;
+import eu.bbmri_eric.negotiator.database.model.DiscoveryServiceSynchronizationJob;
+import eu.bbmri_eric.negotiator.database.model.Organization;
+import eu.bbmri_eric.negotiator.database.model.Resource;
+import eu.bbmri_eric.negotiator.database.repository.AccessFormRepository;
+import eu.bbmri_eric.negotiator.database.repository.DiscoveryServiceRepository;
+import eu.bbmri_eric.negotiator.database.repository.DiscoveryServiceSynchronizationJobRepository;
+import eu.bbmri_eric.negotiator.database.repository.OrganizationRepository;
+import eu.bbmri_eric.negotiator.database.repository.ResourceRepository;
+import eu.bbmri_eric.negotiator.dto.MolgenisBiobank;
+import eu.bbmri_eric.negotiator.dto.MolgenisCollection;
+import eu.bbmri_eric.negotiator.exceptions.EntityNotStorableException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import lombok.extern.apachecommons.CommonsLog;
+import org.hibernate.exception.DataException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientRequestException;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+@CommonsLog
+@Service(value = "DefaultDiscoveryServiceClient")
+public class BBMRIDiscoveryServiceClientImpl implements DiscoveryServiceClient {
+
+  // @Value("${negotiator.molgenis-url}")
+  // private String molgenisURL;
+
+  private BBMRIDiscoveryServiceClientImpl bbmriService = null;
+
+  @Autowired private OrganizationRepository organizationRepository;
+
+  @Autowired private AccessFormRepository accessFormRepository;
+
+  @Autowired private DiscoveryServiceRepository discoveryServiceRepository;
+  @Autowired private ResourceRepository resourceRepository;
+
+  @Autowired
+  private DiscoveryServiceSynchronizationJobRepository discoveryServiceSynchronizationJobRepository;
+
+  private DiscoveryServiceSynchronizationJob jobRecord;
+
+  private WebClient webClient;
+
+  @Autowired
```

Comment:
please remove all autowires

Reply:
There was one not used but why all the others? They are needed as usual in Services